e2eshark/onnx/operators/expand/model.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Advanced Micro Devices
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# run.py creates runmodel.py by concatenating this file model.py
+# and tools/stubs/onnxmodel.py
+# Description: testing Expand
+# See https://onnx.ai/onnx/intro/python.html for intro on creating
+# onnx model using python onnx API
+import numpy, torch, sys
+import onnxruntime
+from onnx import numpy_helper, TensorProto, save_model
+from onnx.helper import (
+    make_model,
+    make_node,
+    make_graph,
+    make_tensor_value_info,
+    make_tensor,
+)
+from onnx.checker import check_model
+
+# import from e2eshark/tools to allow running in current dir, for run through
+# run.pl, commutils is symbolically linked to allow any rundir to work
+sys.path.insert(0, "../../../tools/stubs")
+from commonutils import E2ESHARK_CHECK_DEF
+
+# Create an instance of it for this test
+E2ESHARK_CHECK = dict(E2ESHARK_CHECK_DEF)
+
+# Create an input (ValueInfoProto)
+X = make_tensor_value_info("X", TensorProto.FLOAT, [4, 5])
+
+shape_tensor = make_tensor(
+    name="shape",
+    data_type=TensorProto.INT64,
+    dims=(3,),
+    vals=[3, 4, 5],
+)
+
+shape_node = make_node(
+    "Constant",
+    inputs=[],
+    outputs=["shape"],
+    value=shape_tensor,
+)
+
+# Create an output
+Z = make_tensor_value_info("Z", TensorProto.FLOAT, [3, 4, 5])
```

Comment:
This new test looks similar to these generated tests:

* https://github.com/nod-ai/SHARK-TestSuite/tree/main/iree_tests/onnx/node/generated/test_expand_dim_changed
* https://github.com/nod-ai/SHARK-TestSuite/tree/main/iree_tests/onnx/node/generated/test_expand_dim_unchanged

If you want to modify those test cases, I'd look into the Python sources at https://github.com/onnx/onnx/blob/main/onnx/backend/test/case/node/expand.py and the generation pipeline in ONNX + this repo. The last few PRs to this `onnx/operators/` directory won't scale efficiently to the full list of operators.